fix(linter/plugins): exit early if JS plugins enabled on unsupported platforms

overlookmotel opened the pull request:
JS plugins use raw transfer. Raw transfer does not currently support 32-bit or big-endian platforms.

Exit early with a descriptive error message if `--experimental-js-plugins` option is used on an unsupported platform. This avoids a panic with a weird error later on when first try to create a raw transfer-compatible allocator.

camc314 reviewed (approved):
should we fallback to non-raw transfer at some point?

overlookmotel committed:
fix(linter/plugins): exit early if JS plugins enabled on unsupported platforms (#13689)

JS plugins use raw transfer. Raw transfer does not currently support 32-bit or big-endian platforms.

Exit early with a descriptive error message if `--experimental-js-plugins` option is used on an unsupported platform. This avoids a panic with a weird error later on when first try to create a raw transfer-compatible allocator.
